traitcuration/static/js/github-issue.js

```diff
@@ -0,0 +1,95 @@
+// Function to validate the GitHub issue submission form
+async function validateAndSubmit() {
+  issue_repo = document.querySelector("#id_repo");
+  issue_title = document.querySelector("#id_title");
+  issue_body = document.querySelector("#id_body");
+  isValid = true;
+
+  // Clear any previous errors
+  errorLabels = document.querySelectorAll(".field__error");
+  for (const label of errorLabels) label.innerHTML = "";
+
+  fields = document.querySelectorAll(".field");
+  for (field of fields) field.setCustomValidity("");
+
+  // Make sure the GitHub repo provided is valid and exists
+  try {
+    await axios
+    .get(`https://api.github.com/repos/${issue_repo.value}`, {
```

Comment:
I feel a bit uneasy about making this GET request with an unsanitised user input (`${issue_repo.value}`). Is there a way we could do sanitisation in JS?

Reply:
So regarding this one, if you are referring to converting to HTML encoding to escape non standard symbols I could write something like this:
``` js
function sanitize(string) {
  const map = {
      '&': '&amp;',
      '<': '&lt;',
      '>': '&gt;',
      '"': '&quot;',
      "'": '&#x27;',
      "/": '&#x2F;',
  };
  const reg = /[&<>"'/]/ig;
  return string.replace(reg, (match)=>(map[match]));
}```

However, since all I am making is a GET request to an external API, I'm not sure if it is unsafe. Should I go ahead and use the above `sanitize` function?

Reply:
Since this is a request to an external API made by the user, I agree this is not unsafe in terms of XSS or similar security risk; still, I'd rather avoid accidental nonsensical API requests in case a user badly mistypes something in a field.

To apply this `sanitize` function would a good idea (in the future, it can also be reused throughout the project for similar purposes). A possible alternative is to simply validate GitHub repository name using a regular expression. I would be fine with either of those solutions.